examples/Remarketing/AddRealEstateFeed.php

```diff
@@ -0,0 +1,440 @@
+
+        // Gets the attributes list from the feed and creates a map with keys of each attribute and
+        // values of each corresponding ID.
+        $feedAttributes =
+            iterator_to_array($googleAdsRow->getFeed()->getAttributes()->getIterator());
+
+        $feedPlaceHolderFieldsList = array_map(function (FeedAttribute $feedAttribute) {
+            switch ($feedAttribute->getNameUnwrapped()) {
+                case 'Listing ID':
+                    return RealEstatePlaceholderField::LISTING_ID;
+                    break;
+                case 'Listing Name':
+                    return RealEstatePlaceholderField::LISTING_NAME;
+                    break;
+                case 'Final URLs':
+                    return RealEstatePlaceholderField::FINAL_URLS;
+                    break;
+                case 'Image URL':
+                    return RealEstatePlaceholderField::IMAGE_URL;
+                    break;
+                case 'Contextual Keywords':
+                    return RealEstatePlaceholderField::CONTEXTUAL_KEYWORDS;
+                    break;
+                // Optionally add other real estate placeholder fields.
+                default:
+                    throw new \RuntimeException('Invalid feed attribute name.');
+            }
+        }, $feedAttributes);
+
+        return array_combine($feedPlaceHolderFieldsList, $feedAttributes);
+    }
+
+    /**
+     * Creates a feed mapping for a given feed.
+     *
+     * @param GoogleAdsClient $googleAdsClient the Google Ads API client
+     * @param int $customerId the customer ID
+     * @param string $feedResourceName the feed resource name for creating a feed mapping
+     * @param array $placeHoldersToFeedAttributesMap the map from placeholder fields to feed
+     *      attributes
+     */
+    private static function createFeedMapping(
+        GoogleAdsClient $googleAdsClient,
+        int $customerId,
+        string $feedResourceName,
+        array $placeHoldersToFeedAttributesMap
+    ) {
+        // Maps the feed attribute IDs to the placeholder values. The feed attribute ID is the ID
+        // of the feed attribute created in the createdFeed method. This can be thought of as the
+        // generic ID of the column of the new feed. The placeholder value specifies the type of
+        // column this is in the context of a real estate feed (e.g. a LISTING_ID or
+        // LISTING_NAME). The feed mapping associates the feed column by ID to this type and
+        // controls how the feed attributes are presented in dynamic content.
+        // See RealEstatePlaceholderField.php for the full list of placeholder values.
+        $listingIdMapping = new AttributeFieldMapping([
+            'feed_attribute_id' => new Int64Value([
+                'value' =>
+                    $placeHoldersToFeedAttributesMap[RealEstatePlaceholderField::LISTING_ID]
+                        ->getId()->getValue()
```

Comment:
This can be getIdUnwrapped()
I'm fixing this for flights feed example in my next PR.